### PR TITLE
:package: Move to php-http to replace egeloen/http-adapter

### PR DESCRIPTION
**Pull request by tyx**
- BC Break : php requirement is moved to php 5.5
- BC Break : Remove `RestApiContext::iSendAPostRequestToWithFormData`

### CODE REVIEW

**Comment by tyx** [comment]
Will lead to new major version 5.0

File: src/Rest/RestApiBrowser.php [hunk 1]
```diff
@@ -27,23 +25,20 @@ class RestApiBrowser
...
-        } else {
-            if (is_string($adaptor_name) && HttpAdapterFactory::capable($adaptor_name)) {
-                $this->httpClient = HttpAdapterFactory::create($adaptor_name);
-            } else {
-                $this->httpClient = HttpAdapterFactory::guess();
-            }
-            $this->httpClient->getConfiguration()->setBaseUri($base_url);
-        }
+        $this->host = $host;
+        $this->httpClient = $httpClient ?: HttpClientDiscovery::find();
```
Comment (shouze) [inline src/Rest/RestApiBrowser.php]: What happen if we don't have required any http client in our composer.json? Some exception is throwed?

Comment (tyx) [inline src/Rest/RestApiBrowser.php]: yes

File: src/Rest/RestApiBrowser.php [hunk 2]
```diff
@@ -82,7 +76,7 @@ public function getRequestHeaders()
     }
 
     /**
-     * @return HttpClient
+     * @return Http\Client\HttpClient
      */
     public function getHttpClient()
```
Comment (shouze) [inline src/Rest/RestApiBrowser.php]: Do we still need it? Not used anywhere in the project and we have no reason to expose it once we have some or specified we use some another http client.

**Comment by stephpy** [comment]
:+1:

**Pull request merged**